train_gpt2.c

```diff
@@ -10,14 +10,22 @@ There will be other versions of this code that specialize it and make it fast.
 
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef _WIN32
+#define _USE_MATH_DEFINES
+#endif
 #include <math.h>
 #include <time.h>
 #include <string.h>
-#include <unistd.h>
-#ifdef OMP
+#if defined(OMP) && !(_OPENMP < 202011)
 #include <omp.h>
+#define OMP_ENABLED
+#else
+#pragma message \
```

Comment:
Unfortunately, my version of Visual Studio 2022 is still only shipping with OpenMP 2.0, which is too old to use the syntax of some of your directives, the collapse clause for an example apparently was only added in 5.2 apparently: https://devblogs.microsoft.com/cppblog/open-mp-improvements-in-visual-studio-cpp/#collapse-clause-in-openmp-standard-5-2

Reply:
I have this working (finally) on Windows. Have a number of changes and am hoping you can add your CMake changes to these changes. Should be pushed up over the weekend.